packages/devtools_app/lib/src/trees.dart

```diff
@@ -105,6 +108,28 @@ abstract class TreeNode<T extends TreeNode<T>> {
     child.index = children.length - 1;
   }
 
+  void addChildAtIndex(T child, int index) {
```

Comment:
instead of creating a new method for this, let's add an optional parameter `index` to `addChild`. That way we don't have to duplicate the logic of `child.parent = this;` and `child.index = index;`